src/overlays/actors/ovl_En_Horse_Ganon/z_en_horse_ganon.c

```diff
@@ -25,29 +45,291 @@ const ActorInit En_Horse_Ganon_InitVars = {
+    func_80061ED4(&this->actor.colChkInfo, 0, &collisionCheckInfoInit);
+    func_80A68AC4(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Ganon/func_80A68E14.s")
+void EnHorseGanon_Destroy(EnHorseGanon* this, GlobalContext* globalCtx) {
+    func_800A6888(globalCtx, &this->unk_154);
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+    Collider_DestroyJntSph(globalCtx, &this->colliderSphere);
+}
 
+void func_80A68AC4(EnHorseGanon* this) {
+    this->updateFnIndex = 0;
+    SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, D_80A691C0[0]);
+}
+
+void func_80A68AF0(EnHorseGanon* this, GlobalContext* globalCtx) {
+    this->actor.speedXZ = 0.0f;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+void func_80A68B20(EnHorseGanon* this) {
+    s32 animationChanged;
+    f32 sp30;
+
+    animationChanged = 0;
+    this->updateFnIndex = 1;
+    if (this->actor.speedXZ <= 3.0f) {
+        if (this->currentAnimation != 2) {
+            animationChanged = 1;
+        }
+        this->currentAnimation = 2;
+    } else if (this->actor.speedXZ <= 6.0f) {
+        if (this->currentAnimation != 3) {
+            animationChanged = 1;
+        }
+        this->currentAnimation = 3;
+    } else {
+        if (this->currentAnimation != 4) {
+            animationChanged = 1;
+        }
+        this->currentAnimation = 4;
+    }
+
+    if (this->currentAnimation == 2) {
+        sp30 = this->actor.speedXZ / 3.0f;
+    } else if (this->currentAnimation == 3) {
+        sp30 = this->actor.speedXZ / 5.0f;
+        Audio_PlaySoundGeneral(NA_SE_EV_HORSE_RUN, &this->actor.unk_E4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    } else if (this->currentAnimation == 4) {
+        sp30 = this->actor.speedXZ / 7.0f;
+        Audio_PlaySoundGeneral(NA_SE_EV_HORSE_RUN, &this->actor.unk_E4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    } else {
+        sp30 = 1.0f;
+    }
+
+    if (animationChanged == 1) {
+        SkelAnime_ChangeAnim(&this->skelAnime, D_80A691B0[this->currentAnimation],
+                             animPlaybackSpeed[this->currentAnimation] * sp30 * 1.5f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_80A691B0[this->currentAnimation]->genericHeader), 2, -3.0f);
+    } else {
+        SkelAnime_ChangeAnim(&this->skelAnime, D_80A691B0[this->currentAnimation],
+                             animPlaybackSpeed[this->currentAnimation] * sp30 * 1.5f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_80A691B0[this->currentAnimation]->genericHeader), 2, 0.0f);
+    }
+}
+
+void func_80A68DB0(EnHorseGanon* this, GlobalContext* globalCtx) {
+    if (this->currentAnimation == 2) {
+        func_80A68870(this);
+    }
+
+    func_80A686A8(this, globalCtx);
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        func_80A68B20(this);
+    }
+}
+
+void func_80A68E14(EnHorseGanon* this, GlobalContext* globalCtx) {
+    u32 junk; // aligns stack properly
```

Comment:
we generally call this pad, you can add numbers to it if theres more than one stack padding variable. no need for the comment as well, the name pad will give that away